src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -96,10 +99,24 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
         String renderedMessage = renderer.render(document);
         
         String result = EmojiParser.parseToUnicode(renderedMessage);	
-		String resultDecimal = EmojiParser.parseToHtmlDecimal(result);	  	
-    %>
-      <li><strong><%= author %>:</strong> <%= resultDecimal %> </li>
-    <%
+		String resultDecimal = EmojiParser.parseToHtmlDecimal(result);
+
+        //uses original str to validate url because the other strings contain extra stuff
+		if (urlValidator.isValid(str) && ( str.endsWith(".png") || str.endsWith(".jpg") || str.endsWith(".gif") ) ) {
```

Comment:
There might be a couple corner cases with images that end with, say, `.jpeg` or `.JPG`, but let's not worry about those for now. This looks great!

Reply:
Okay